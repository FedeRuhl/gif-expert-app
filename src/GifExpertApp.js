// import PropTypes from 'prop-types';
// import {useState} from 'react';

import { useState } from "react";
import AddCategory from "./Components/AddCategory";
import GifGrid from "./Components/GifGrid";

const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h1>Gif expert app</h1>
            <AddCategory setCategories={ setCategories }/>
            <hr/>

            {
                <ol>
                    {
                        categories.map(category => {
                            return <GifGrid
                                key={ category }
                                category={ category }
                            />
                        })
                    }
                </ol>
            }
        </>
    );
}

export default GifExpertApp;