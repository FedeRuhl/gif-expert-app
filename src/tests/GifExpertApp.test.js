import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';
import React from 'react';

describe('Gif Expert App component test', () => {

    test('should show the componente correctly', () => {
        const wrapper = shallow(<GifExpertApp />);
       expect(wrapper).toMatchSnapshot(); 
    });

    test('should show a categories list', () => {
        const defaultCategories = ['Samurai X', 'Maradona'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ defaultCategories } />);

        expect(wrapper.find('GifGrid').length).toBe(defaultCategories.length);
    });
    
    
});