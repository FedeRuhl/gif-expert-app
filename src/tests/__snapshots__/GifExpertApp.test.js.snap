// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif Expert App component test should show the componente correctly 1`] = `
<Fragment>
  <h1>
    Gif expert app
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
